client/network/src/discovery.rs

```diff
@@ -949,7 +920,11 @@ mod tests {
 					config.finish()
 				};
 
-				let mut swarm = Swarm::new(transport, behaviour, keypair.public().to_peer_id());
+				let mut swarm = Swarm::with_threadpool_executor(
```

Comment:
I've just merged https://github.com/paritytech/substrate/pull/12646 (finally), so, ideally, this and others should be replaced with a tokio executor.

```suggestion
				let mut swarm = Swarm::with_tokio_executor(
```

Reply:
And, may be, we should even use `Swarm::with_executor()` and pass the spawner from the network params https://github.com/paritytech/substrate/blob/2bde8c1a44d8e376e3e6e5d2dd7266124ff5efa2/client/network/src/config.rs#L70
(But I don't know whether it does anything useful apart of additional debug logging.)

Reply:
why it's not desirable to use threadpool executor in tests?

Reply:
https://github.com/paritytech/substrate/pull/12734/commits/edd7d8d60a47770edd562d9d24fbdcb802faad85

Reply:
I missed that this is just a test, so I apologize for not paying enough attention — there should be no harm of using a threadpool executor here. On the other hand, as per https://github.com/libp2p/rust-libp2p/issues/3107, it looks like `ThreadPool` executor is going to be deprecated in libp2p completely, so switching to tokio in advance might make sense.

Reply:
Thanks for tagging the above issue 🙏  Didn't know they are going to deprecate `ThreadPool`.